test/built-ins/TypedArray/prototype/copyWithin/coerced-values-end-detached-prototype.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2019 Google. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.copywithin
+description: >
+  SECURITY: end argument is coerced to an integer values
+  causing array detachment, but the value is still defined
+  by a prototype
+info: |
+  22.2.3.5%TypedArray%.prototype.copyWithin ( target, start [ , end ] )
+  ...
+  8. If end is undefined, let relativeEnd be len; else let relativeEnd be ? ToInteger(end).
+  ... 
+  10. Let count be min(final - from, len - to).
+  11. If count > 0, then
+    a. NOTE: The copying must be performed in a manner that preserves the bit-level encoding of the source data.
+    b. Let buffer be O.[[ViewedArrayBuffer]].
+    c. If IsDetachedBuffer(buffer) is true, throw a TypeError exception. 
+  ...
+includes: [testTypedArray.js, detachArrayBuffer.js]
+features: [TypedArray]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  
+  var ta;
+  var array = [];
+
+  function detachAndReturnIndex(){
+      $DETACHBUFFER(ta.buffer);
+      Object.setPrototypeOf(ta, array);
+      return 101;
```

Comment:
Interesting that setting the end being 1 number after start (`100` at line 39) makes it detect an implementation bug on V8.